pre-docs/writing_tasks.md

```diff
@@ -708,6 +708,51 @@ else
 fi
 ```
 
+If you intend to use this task with PE and assign RBAC permissions for it make
+sure the script safely handles parameters or validate them to prevent shell
+injection vulnerabilities.
+
+### Wrapping an existing script
+
+If the script is not already installed on targets and you do not want to edit
+it, for example if it's a script someone else maintains, you can wrap the
+script as is with it a small task.
+
+Given a script that accepts 2 positional args `filename` and `version` called `myscript.sh`:
+
+1. Copy the script to the module's `files/` directory.
+
+2. Create a metadata file for the task that includes the parameters and file dependency.
+
+   ```
+   {
+     "input_method": "environment",
+     "parameters": {
+       "filename": { "type": "String[1]" },
+       "version": { "type": "String[1]" }
+     },
+     "files": [ "script_example/files/myscript.sh" ]
+   }
+   ```
+
+3. Create a small wrapper task that reads environment variables and calls the task.
+
+   ```
+   #!/usr/bin/env bash
+   set -e
+
+   script_file="$PT__installdir/script_example/files/myscript.sh"
+   # If this task is going to be run from windows nodes the wrapper must make sure it's exectutable
+   chmod +x $script_file
```

Comment:
I don't think what you have here is contains any injection vulnerabilities in the parameters that are passed. 

Tested that stderr from script is also visible:
```
#!/usr/bin/env bash
echo 'in myscript'
args=("$@")
echo ${args[0]}
echo ${args[1]}
(>&2 echo "std error from myscript")
```
```
cas@cas-ThinkPad-T460p:~/working_dir/bolt/bolt-modules/boltlib$ bolt task run test::script_wrapper filename='`rm /root/test.txt`' version='(rm /root/test.txt)' -n jhkyqbnchikhkc4.delivery.puppetlabs.net
Started on jhkyqbnchikhkc4.delivery.puppetlabs.net...
Finished on jhkyqbnchikhkc4.delivery.puppetlabs.net:
  in myscript
  `rm
  /root/test.txt`
  std error from myscript
  {
  }
Successful on 1 node: jhkyqbnchikhkc4.delivery.puppetlabs.net
Ran on 1 node in 0.78 seconds
```

Reply:
Actually after talking with about it with @sensei-de-loafage he showed me a way to control word splitting such that if there were spaces in the arguments those would not be split on space. 

```
#!/usr/bin/env bash
set -e
script_file="$PT__installdir/script_example/files/myscript.sh"
# If this task is going to be run from windows nodes the wrapper must make sure it's executable
chmod +x $script_file
commandline=("$script_file" "$PT_filename" "$PT_version")
"${commandline[@]}" 2>&
```
Example showing that the whole line is passed as first arg to script even through it has spaces:
```
cas@cas-ThinkPad-T460p:~/working_dir/bolt$ bolt task run test::script_wrapper filename="\`rm -f /root/test.txt\`" version=version -n xne1m3qhr08la58.delivery.puppetlabs.net
Started on xne1m3qhr08la58.delivery.puppetlabs.net...
Finished on xne1m3qhr08la58.delivery.puppetlabs.net:
  in myscript
  `rm -f /root/test.txt`
  version
  std error from myscript
  {
  }
Successful on 1 node: xne1m3qhr08la58.delivery.puppetlabs.net
Ran on 1 node in 1.56 seconds
```